### PR TITLE
[WIP] [PRO-1913] Authed namespace scope

### PR DESCRIPTION
**Pull request by jerrytrieu**

### CODE REVIEW

File: core/src/main/scala/org/genivi/sota/core/DeviceUpdatesResource.scala [hunk 1]
```diff
@@ -44,8 +46,8 @@ import scala.concurrent.Future
 class DeviceUpdatesResource(db: Database,
                             resolverClient: ExternalResolverClient,
                             deviceRegistry: DeviceRegistry,
-                            authNamespace: Directive1[Namespace],
-                            authDirective: AuthScope => Directive0,
+                            authNamespace: Directive1[AuthedNamespaceScope],
+                            authDirective: (AuthedNamespaceScope, AuthScope, Boolean) => Directive0,
```
Comment (Danten) [inline core/src/main/scala/org/genivi/sota/core/DeviceUpdatesResource.scala]: authDirective is not used anymore

File: device-registry/src/main/scala/org/genivi/sota/device_registry/SystemInfoResource.scala [hunk 2]
```diff
@@ -14,13 +14,15 @@ import org.genivi.sota.device_registry.db._
...
+import org.genivi.sota.http.AuthedNamespaceScope
 import org.genivi.sota.marshalling.CirceMarshallingSupport._
 import org.slf4j.LoggerFactory
 import slick.driver.MySQLDriver.api._
 
 import scala.concurrent.{ExecutionContext, Future}
 
-class SystemInfoResource(authDirective: AuthScope => Directive0, deviceNamespaceAuthorizer: Directive1[Uuid])
+class SystemInfoResource(authNamespace: Directive1[AuthedNamespaceScope],
+    authDirective: (AuthedNamespaceScope, AuthScope, Boolean) => Directive0, deviceNamespaceAuthorizer: Directive1[Uuid])
```
Comment (Danten) [inline device-registry/src/main/scala/org/genivi/sota/device_registry/SystemInfoResource.scala]: Same here, don't need ```authDirective```

File: common/src/main/scala/org/genivi/sota/http/AuthNamespaceDirectives.scala [hunk 3]
```diff
@@ -80,15 +99,16 @@ object AuthNamespaceDirectives {
...
-                   (implicit nsFromToken: NsFromToken[T], decoder: Decoder[T]): Directive1[Namespace] =
+                   (implicit nsFromToken: NsFromToken[T], decoder: Decoder[T]): Directive1[AuthedNamespaceScope] =
     extractCredentials flatMap { creds =>
       val maybeNamespace = creds match {
         case Some(OAuth2BearerToken(serializedToken)) =>
           NsFromToken.parseToken[T](serializedToken).flatMap{ token =>
+            val authedNs = nsFromToken.toNamespaceScope(token)
             ns0 match {
-              case Some(ns) if nsFromToken.accept(ns, token) => Xor.right(ns)
+              case Some(ns) if authedNs.hasNamespace(ns) => Xor.right(authedNs)
```
Comment (jerrytrieu) [inline common/src/main/scala/org/genivi/sota/http/AuthNamespaceDirectives.scala]: Yes, right.

**Comment by jerrytrieu** [comment]
Completed by #678

**Pull request closed without merging**